src/Discord.Net.WebSocket/Entities/Guilds/SocketGuild.cs

```diff
@@ -518,29 +518,37 @@ internal AudioInStream GetAudioStream(ulong userId)
                 promise = new TaskCompletionSource<AudioClient>();
                 _audioConnectPromise = promise;
 
-                if (_audioClient == null)
+                if (!external)
                 {
-                    var audioClient = new AudioClient(this, Discord.GetAudioId(), channelId);
-                    audioClient.Disconnected += async ex =>
+                    if (_audioClient == null)
                     {
-                        if (!promise.Task.IsCompleted)
+                        var audioClient = new AudioClient(this, Discord.GetAudioId(), channelId);
+                        audioClient.Disconnected += async ex =>
                         {
-                            try { audioClient.Dispose(); } catch { }
-                            _audioClient = null;
-                            if (ex != null)
-                                await promise.TrySetExceptionAsync(ex);
-                            else
-                                await promise.TrySetCanceledAsync();
-                            return;
-                        }
-                    };
-                    audioClient.Connected += () =>
-                    {
-                        var _ = promise.TrySetResultAsync(_audioClient);
-                        return Task.Delay(0);
-                    };
-                    configAction?.Invoke(audioClient);
-                    _audioClient = audioClient;
+                            if (!promise.Task.IsCompleted)
+                            {
+                                try
+                                { audioClient.Dispose(); }
+                                catch { }
+                                _audioClient = null;
+                                if (ex != null)
+                                    await promise.TrySetExceptionAsync(ex);
+                                else
+                                    await promise.TrySetCanceledAsync();
+                                return;
+                            }
+                        };
+                        audioClient.Connected += () =>
+                        {
+                            var _ = promise.TrySetResultAsync(_audioClient);
+                            return Task.Delay(0);
+                        };
+                        configAction?.Invoke(audioClient);
```

Comment:
Is this even necessary anymore? I think we might be able to remove it without breaking any existing code, even if it would be a breaking change.